File: cifrado/escritorio/codigo/c++/cesar.cpp
#include "cesar.h"

Cesar::Cesar(Texto *texto,int semilla){
    abecedario = new Abecedario();
    this->texto = texto;
    mensajeCifrado = "";
    posicion = semilla;
    mensajeCifrado = cifrar();
}

string Cesar::cifrar(){
    string tC = "";
    while (texto->hayMasTexto())
    {
        char letra = texto->getLetraMayuscula();
        if ((int)letra != 0)
        {
            tC += abecedario->desplazarLetraEn(letra, posicion);
        }
    }
    return tC;
}


string Cesar::verMensaje(){
    return mensajeCifrado;
}

File: cifrado/escritorio/codigo/ruby/Cesar.rb

class Cesar
    @Texto
    def initialize(entrada,desplazamiento)
        @Texto=cifrar(entrada,desplazamiento)
    end 

    def cifrar(texto,semilla)
        salida=""
        texto.length.times do |index| 
            if /[a-z]/.match(texto[index])
              letter = texto[index].ord + semilla
              if letter > "z".ord
                salida.concat((letter - 26).chr)
              else
                salida.concat(letter.chr)
              end
            elsif /[A-Z]/.match(texto[index])
              letter = texto[index].ord + semilla
              if letter > "Z".ord
                salida.concat((letter - 26).chr)
              else
                salida.concat(letter.chr)
              end
            else
                salida.concat(texto[index])
            end
          end
          return salida
    end


    def getMensaje()
        @Texto
    end
end

